pkgs/cupertino_http/lib/cupertino_http.dart

```diff
@@ -697,6 +712,28 @@ void _setupDelegation(
           forwardedData.finish();
         }
         break;
+      case ncb.MessageType.FinishedDownloading:
+        final finishedDownloading =
+            ncb.CUPHTTPForwardedFinishedDownloading.castFrom(forwardedDelegate);
+        try {
+          if (onFinishedDownloading == null) {
+            break;
+          }
+          try {
```

Comment:
What is the meaning of outer try-finally? it doesn't look like `if (onFinishedDownloading == null)` could throw.

Reply:
It can't. I just want to ensure that `finishedDownloading.finish()` is called because, if it isn't, then `cupertino_http` will deadlock.